bootstrap/kubeadm/internal/webhooks/kubeadmconfigtemplate.go

```diff
@@ -44,13 +46,20 @@ func (webhook *KubeadmConfigTemplate) SetupWebhookWithManager(mgr ctrl.Manager)
 type KubeadmConfigTemplate struct{}
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type.
-func (webhook *KubeadmConfigTemplate) Default(_ context.Context, obj runtime.Object) error {
+func (webhook *KubeadmConfigTemplate) Default(ctx context.Context, obj runtime.Object) error {
 	c, ok := obj.(*bootstrapv1.KubeadmConfigTemplate)
 	if !ok {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected a KubeadmConfigTemplate but got a %T", obj))
 	}
 
-	c.Spec.Template.Spec.Default()
+	req, err := admission.RequestFromContext(ctx)
+	if err != nil {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected an admission.Request inside context: %v", err))
+	}
+
+	if topology.IsDryRunRequest(req, c) {
```

Comment:
What about adding something like
```suggestion
	// In case of dry run request from the topology controller, apply defaults from older versions of CAPI
	// so we do not trigger rollouts when dealing with objects created before dropping those defaults.
	if topology.IsDryRunRequest(req, c) {
```